// node-pixels/dist/pixels-lib/colour_tints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.red_effect = exports.purplescale_imgdata = exports.rosetint_imgdata = exports.twenties_imgdata = exports.evening_imgdata = exports.mellow_imgdata = exports.haze_imgdata = exports.pink_aura_imgdata = exports.solange_dark_imgdata = exports.zapt_imgdata = exports.eon_imgdata = exports.neue_imgdata = exports.radio_imgdata = exports.ocean_imgdata = exports.greengreyscale_imgdata = exports.crimson = exports.phase = exports.grime = exports.redgreyscale_imgdata = exports.warmth = exports.slate_imgdata = exports.serenity_imgdata = exports.vintage_imgdata = exports.perfume_imgdata = exports.greyscale_imgdata = exports.frontward_imgdata = exports.coral_imgdata = exports.lemon_imgdata = exports.sunset = exports.wood = exports.lix_conv = exports.ryo_conv = exports.solange_imgdata = exports.solange_grey_imgdata = exports.aeon_imgdata = exports.blue_greyscale_imgdata = exports.cool_twilight_imgdata = exports.blues_imgdata = void 0;
let i;
const helpers_1 = require("./helpers");
const lemon_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = imgData.data[i] + 50;
    }
    return imgData;
};
exports.lemon_imgdata = lemon_imgdata;
const coral_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 2] = imgData.data[i + 1] + 50;
    }
    return imgData;
};
exports.coral_imgdata = coral_imgdata;
const frontward_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i + 2];
        imgData.data[i + 2] = imgData.data[i + 1] + 50;
    }
    return imgData;
};
exports.frontward_imgdata = frontward_imgdata;
const vintage_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 120;
        imgData.data[i + 1] += 70;
        imgData.data[i + 2] += 13;
    }
    return imgData;
};
exports.vintage_imgdata = vintage_imgdata;
const perfume_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 80;
        imgData.data[i + 1] += 40;
        imgData.data[i + 2] += 120;
    }
    return imgData;
};
exports.perfume_imgdata = perfume_imgdata;
const serenity_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 10;
        imgData.data[i + 1] += 40;
        imgData.data[i + 2] += 90;
    }
    return imgData;
};
exports.serenity_imgdata = serenity_imgdata;
const pink_aura_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 90;
        imgData.data[i + 1] += 10;
        imgData.data[i + 2] += 90;
    }
    return imgData;
};
exports.pink_aura_imgdata = pink_aura_imgdata;
const haze_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 90;
        imgData.data[i + 1] += 90;
        imgData.data[i + 2] += 10;
    }
    return imgData;
};
exports.haze_imgdata = haze_imgdata;
const mellow_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 2] = 120 - imgData.data[i + 2];
    }
    return imgData;
};
exports.mellow_imgdata = mellow_imgdata;
const solange_dark_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = 200 - imgData.data[i];
        // imgData.data[i + 1] = 255 - imgData.data[i + 1];
        // imgData.data[i + 2] = 255 - imgData.data[i + 2];
    }
    return imgData;
};
exports.solange_dark_imgdata = solange_dark_imgdata;
const zapt_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        // imgData.data[i] = 255 - imgData.data[i];
        imgData.data[i + 1] = 255 - imgData.data[i + 1];
        // imgData.data[i + 2] = 255 - imgData.data[i + 2];
    }
    return imgData;
};
exports.zapt_imgdata = zapt_imgdata;
const neue_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 2] = 255 - imgData.data[i + 2];
        imgData.data[i] = imgData.data[i] + 20;
    }
    return imgData;
};
exports.neue_imgdata = neue_imgdata;
const eon_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = 120 - imgData.data[i + 1];
        imgData.data[i + 2] = 100 - imgData.data[i + 2];
    }
    return imgData;
};
exports.eon_imgdata = eon_imgdata;
const aeon_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = 60 - imgData.data[i + 1];
        imgData.data[i + 2] = 100 - imgData.data[i + 2];
    }
    return imgData;
};
exports.aeon_imgdata = aeon_imgdata;
const rosetint_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 80;
        imgData.data[i + 1] = avg + 20;
        imgData.data[i + 2] = avg + 31;
    }
    return imgData;
};
exports.rosetint_imgdata = rosetint_imgdata;
const slate_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 4;
        imgData.data[i + 1] = avg + 3;
        imgData.data[i + 2] = avg + 12;
    }
    return imgData;
};
exports.slate_imgdata = slate_imgdata;
const purplescale_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 90;
        imgData.data[i + 1] = avg + 40;
        imgData.data[i + 2] = avg + 80;
    }
    return imgData;
};
exports.purplescale_imgdata = purplescale_imgdata;
const radio_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 5;
        imgData.data[i + 1] = avg + 40;
        imgData.data[i + 2] = avg + 20;
    }
    return imgData;
};
exports.radio_imgdata = radio_imgdata;
const twenties_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 18;
        imgData.data[i + 1] = avg + 12;
        imgData.data[i + 2] = avg + 20;
    }
    return imgData;
};
exports.twenties_imgdata = twenties_imgdata;
const ocean_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] += 10;
        imgData.data[i + 1] += 20;
        imgData.data[i + 2] += 90;
    }
    return imgData;
};
exports.ocean_imgdata = ocean_imgdata;
const greyscale_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg;
        imgData.data[i + 1] = avg;
        imgData.data[i + 2] = avg;
    }
    return imgData;
};
exports.greyscale_imgdata = greyscale_imgdata;
const redgreyscale_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 100;
        imgData.data[i + 1] = avg + 40;
        imgData.data[i + 2] = avg + 20;
    }
    return imgData;
};
exports.redgreyscale_imgdata = redgreyscale_imgdata;
const greengreyscale_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 20;
        imgData.data[i + 1] = avg + 70;
        imgData.data[i + 2] = avg + 20;
    }
    return imgData;
};
exports.greengreyscale_imgdata = greengreyscale_imgdata;
const warmth = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i] + 10;
        imgData.data[i + 1] = imgData.data[i + 1] + 18;
    }
    return imgData;
};
exports.warmth = warmth;
const crimson = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i] + 20;
        imgData.data[i + 1] = imgData.data[i + 2] + 20;
    }
    return imgData;
};
exports.crimson = crimson;
const phase = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i] + (0, helpers_1.getRandomNumber)(10, 20);
        imgData.data[i + 1] = imgData.data[i + 2] + (0, helpers_1.getRandomNumber)(10, 20);
        imgData.data[i + 2] = imgData.data[i + 2] + (0, helpers_1.getRandomNumber)(10, 20);
    }
    return imgData;
};
exports.phase = phase;
const grime = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = imgData.data[i] + 5;
        imgData.data[i] = imgData.data[i] + 1;
    }
    return imgData;
};
exports.grime = grime;
const evening_imgdata = (imgData) => {
    let SAT_ADJ = 60;
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] -= SAT_ADJ;
        imgData.data[i + 1] -= SAT_ADJ;
        imgData.data[i + 2] -= SAT_ADJ;
    }
    return imgData;
};
exports.evening_imgdata = evening_imgdata;
const sunset = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = imgData.data[i] + 50;
        imgData.data[i + 2] = imgData.data[i + 2] + 12;
    }
    return imgData;
};
exports.sunset = sunset;
const wood = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i] + 30;
        imgData.data[i + 1] = imgData.data[i + 1] + 12;
    }
    return imgData;
};
exports.wood = wood;
const lix_conv = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = 255 - imgData.data[i];
        imgData.data[i + 1] = 255 - imgData.data[i + 1];
    }
    return imgData;
};
exports.lix_conv = lix_conv;
const ryo_conv = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = 255 - imgData.data[i];
        imgData.data[i + 2] = 255 - imgData.data[i + 2];
    }
    return imgData;
};
exports.ryo_conv = ryo_conv;
const blue_greyscale_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        let avg = (imgData.data[i] + imgData.data[i + 1] + imgData.data[i + 2]) / 3;
        imgData.data[i] = avg + 20;
        imgData.data[i + 1] = avg + 30;
        imgData.data[i + 2] = avg + 60;
    }
    return imgData;
};
exports.blue_greyscale_imgdata = blue_greyscale_imgdata;
const solange_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = 255 - imgData.data[i];
    }
    return imgData;
};
exports.solange_imgdata = solange_imgdata;
const solange_grey_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i + 2];
        imgData.data[i + 1] = imgData.data[i + 1];
        imgData.data[i + 2] = imgData.data[i];
    }
    return imgData;
};
exports.solange_grey_imgdata = solange_grey_imgdata;
const cool_twilight_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = 255 - imgData.data[i + 1];
        imgData.data[i + 2] = imgData.data[i + 2] + 70;
    }
    return imgData;
};
exports.cool_twilight_imgdata = cool_twilight_imgdata;
const blues_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 2] = 255 - imgData.data[i + 2];
    }
    return imgData;
};
exports.blues_imgdata = blues_imgdata;
const red_effect = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = imgData.data[i] + 200;
        imgData.data[i + 1] = imgData.data[i + 1] - 50;
        imgData.data[i + 2] = imgData.data[i + 2] * 0.5;
    }
    return imgData;
};
exports.red_effect = red_effect;
