// node-pixels/dist/pixels-lib/gamma.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.red_gamma = exports.green_gamma = exports.bluered_gamma = exports.yellow_gamma = exports.purple_gamma = exports.teal_gamma = exports.gamma = void 0;
let i;
// These gamma functions provide a slight darkening effect and a colour tint.
const gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = Math.pow(imgData.data[i] / 255, 5) * 255;
        imgData.data[i + 1] = Math.pow(imgData.data[i + 1] / 255, 5) * 255;
        imgData.data[i + 2] = Math.pow(imgData.data[i + 2] / 255, 5) * 255;
    }
    return imgData;
};
exports.gamma = gamma;
const teal_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = Math.pow(imgData.data[i] / 255, 5) * 255;
    }
    return imgData;
};
exports.teal_gamma = teal_gamma;
const purple_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = Math.pow(imgData.data[i + 1] / 255, 5) * 255;
    }
    return imgData;
};
exports.purple_gamma = purple_gamma;
const yellow_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 2] = Math.pow(imgData.data[i + 2] / 255, 5) * 255;
    }
    return imgData;
};
exports.yellow_gamma = yellow_gamma;
const bluered_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = Math.pow(imgData.data[i] / 255, 5) * 255;
        imgData.data[i + 1] = Math.pow(imgData.data[i + 1] / 255, 5) * 255;
    }
    return imgData;
};
exports.bluered_gamma = bluered_gamma;
const green_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i] = Math.pow(imgData.data[i] / 255, 5) * 255;
        imgData.data[i + 2] = Math.pow(imgData.data[i + 2] / 255, 5) * 255;
    }
    return imgData;
};
exports.green_gamma = green_gamma;
const red_gamma = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i + 1] = Math.pow(imgData.data[i + 1] / 255, 5) * 255;
        imgData.data[i + 2] = Math.pow(imgData.data[i + 2] / 255, 5) * 255;
    }
    return imgData;
};
exports.red_gamma = red_gamma;
