// node-pixels/dist/pixels-lib/offset_filters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.rgbSplit = exports.offset_blue_imgdata = exports.offset_green_imgdata = exports.extreme_offset_red = exports.extreme_offset_green = exports.extreme_offset_blue = exports.extra_offset_red = exports.extra_offset_green = exports.extra_offset_blue = exports.offset = void 0;
let i;
const extreme_offset_blue = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 35;
        imgData.data[i + 2] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extreme_offset_blue = extreme_offset_blue;
const extra_offset_blue = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 15;
        imgData.data[i + 2] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extra_offset_blue = extra_offset_blue;
const extreme_offset_green = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 35;
        imgData.data[i + 1] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extreme_offset_green = extreme_offset_green;
const extra_offset_green = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 15;
        imgData.data[i + 1] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extra_offset_green = extra_offset_green;
const extreme_offset_red = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 35;
        imgData.data[i] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extreme_offset_red = extreme_offset_red;
const extra_offset_red = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 15;
        imgData.data[i] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.extra_offset_red = extra_offset_red;
const offset = (imgData) => {
    for (let i = 0; i < imgData.data.length; i += 4) {
        var offset = 5;
        imgData.data[i] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.offset = offset;
const offset_green_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 5;
        imgData.data[i + 1] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.offset_green_imgdata = offset_green_imgdata;
const offset_blue_imgdata = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        var offset = 5;
        imgData.data[i + 2] = imgData.data[i + 4 * offset * offset] == undefined ? 0 : imgData.data[i + 4 * offset];
    }
    return imgData;
};
exports.offset_blue_imgdata = offset_blue_imgdata;
const rgbSplit = (imgData) => {
    for (i = 0; i < imgData.data.length; i += 4) {
        imgData.data[i - 150] = imgData.data[i + 0];
        imgData.data[i + 500] = imgData.data[i + 1];
        imgData.data[i - 300] = imgData.data[i + 2];
    }
    return imgData;
};
exports.rgbSplit = rgbSplit;
